lib/lambda_ethereum_consensus/p2p/block_downloader.ex

```diff
@@ -126,22 +125,18 @@ defmodule LambdaEthereumConsensus.P2P.BlockDownloader do
     end
   end
 
-  @spec parse_response(binary, integer) ::
+  @spec parse_response(binary) ::
           {:ok, [binary()]} | {:error, binary()}
-  def parse_response(response_chunk, count) do
+  def parse_response(response_chunk) do
     fork_context = @fork_context
 
     case response_chunk do
       <<>> ->
         {:error, "unexpected EOF"}
 
       <<0, ^fork_context::binary-size(4)>> <> rest ->
-        result = rest |> :binary.split(<<0, fork_context::binary-size(4)>>, [:global])
-
-        case result do
-          chunks when length(chunks) == count -> {:ok, chunks}
-          _ -> {:error, "expected #{count} chunks, got #{length(result)}"}
-        end
+        chunks = rest |> :binary.split(<<0, fork_context::binary-size(4)>>, [:global])
+        {:ok, chunks}
```

Comment:
Do we still check the length somewhere else?

Reply:
no, it doesn't really make sense in the context of blocks by range since you could have an empty slot. It might make more sense in the context of blocks by root

Reply:
I thought it would just return an error code in the corresponding response chunk... TIL